front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -144,37 +182,84 @@ const Network = ({
       ctx.textAlign = 'center';
       ctx.textBaseline = 'middle';
       ctx.fillStyle = nodeColors[node.color_index % nodeColors.length];
+      const textWidth = ctx.measureText(label).width;
+      node.nodeVerticalOffset = nodeVerticalOffset;
+      node.textWidth = textWidth;
+      node.globalScale = globalScale;
+      node.nodeFontSize = nodeFontSize;
 
       if (globalScale >= visibleKeywordLabelScale) {
         ctx.fillText(label, node.x, nodeVerticalOffset);
+
+        if (node == hoverNode) {
+          drawHideIcon(ctx, node);
+
+          const [offsetX, offsetY] = pointerPosition;
+          const rect = drawHideIconClickBox(node);
+
+          if (ctx.isPointInPath(rect, offsetX, offsetY)) {
+            tooltipRef.textContent = formatMessage(messages.networkHideNode);
+          } else {
+            tooltipRef.textContent = label;
+          }
+        }
       }
     }
   };
 
-  const handleNodeClick = (node: Node) => {
-    const keywords =
-      query.keywords && typeof query.keywords === 'string'
-        ? [query.keywords]
-        : query.keywords;
-
-    clHistory.replace({
-      pathname,
-      search: stringify(
-        // Toggle selected keywords in url
-        {
-          ...query,
-          keywords: keywords
-            ? !keywords.includes(node.id)
-              ? [keywords, node.id]
-              : keywords.filter((keyword: string) => keyword !== node.id)
-            : node.id,
-        },
-        { addQueryPrefix: true, indices: false }
-      ),
-    });
+  const getURLArrayParam = (key: string) => {
+    const value = searchParams.getAll(key);
+    return typeof value === 'string' ? [value] : value ? value : [];
+  };
+
+  const getKeywords = ({ id }: Node, isHiding: boolean) => {
+    let keywords = getURLArrayParam('keywords');
+
+    if (keywords.includes(id)) {
+      keywords = keywords.filter((keyword: string) => keyword !== id);
+    } else if (!isHiding) {
+      keywords.push(id);
+    }
+    return keywords;
+  };
+
+  const handleNodeClick = (node: Node, event) => {
+    let isHiding = false;
+    const hidden_keywords = getURLArrayParam('hidden_keywords');
+    if (node.x && node.y) {
```

Comment:
I don't know if `node.x` and `node.y` can be 0, but if so may be better to do `node.x !== undefined && node.y !== undefined`

Reply:
good point